.github/actions/bundle-schema/action.yml

```diff
@@ -0,0 +1,53 @@
+name: Bundle & Push JSON schema
+description: 'Bundle & Push JSON schema'
+author: aica-technology
+inputs:
+  schema-path:
+    description: "The path to the schema"
+    required: true
+  schema:
+    description: "The name of the schema"
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+    # Check if the version has been updated
+    - id: versions
+      run: |
+        NEW_VERSION=$(jq -r '.["$id"]' ${{ inputs.schema-path }})
+        echo "new_version=${NEW_VERSION}" >> $GITHUB_OUTPUT
+        IFS='/' read -a id_array <<< "$NEW_VERSION"
+        echo "version_only=${id_array[1]}" >> $GITHUB_OUTPUT
+        git checkout HEAD^
+        PREV_VERSION=$( jq -r '.["$id"]' ${{ inputs.schema-path }})
+        echo "prev_version=${PREV_VERSION}" >> $GITHUB_OUTPUT
+      shell: bash
+    - uses: actions/checkout@v3
+    # Bundle JSON schema
+    - uses: actions/setup-node@v3
+      with:
+        node-version: "18"
+        cache: "npm"
+        cache-dependency-path: "utils/package-lock.json"
+    - name: Install dependencies
+      shell: bash
+      run: |
+        cd utils
+        npm ci
+        pip3 install javascript
+    - name: Bundle & Push schemas
+      if: ${{ steps.versions.outputs.new_version != steps.versions.outputs.prev_version }}
+      shell: bash
+      run: |
+        python3 utils/bundle.py ${{ inputs.schema-path }} ${{ inputs.schema }}
+        mkdir -p docs/static/schemas/${{ steps.versions.outputs.version_only }}
+        mv ${{ inputs.schema }} docs/static/schemas${{ steps.versions.outputs.new_version }}
+        git config --global user.name github-actions[bot]
+        git config --global user.email github-actions[bot]@users.noreply.github.com
+        git add .
+        git commit -m "docs: ${{ inputs.schema }} ${{ steps.versions.outputs.version_only }}"
+        git push
```

Comment:
It pushes on the same branch?

Reply:
yes it pushes on the same branch. There's a condition for the workflow:
`
on:
  push:
    branches:
      - main
`
so I think the workflow happens on the main branch and this push will push the changes to main, maybe I'm wrong? @LouisBrunner

Reply:
Yes, that's correct! I missed the `on` condition, thanks.

I don't like when the CI pushes on branches used by humans as it is a pain to use (have to pull everytime you push, etc) but it's a good v1 and we are happy with the downsides for now I think.

Reply:
yes I think the benefits of this approach outweigh the downsides and it's a good v1